sonic-net/sonic-buildimage
# [build/frr] Properly reset sonic-frr/frr branch after build

#### Why I did it

After a build ./src/sonic-frr/frr was always marked as "modified". This changeset resets it to the way it was before the build.

#### How I did it

The sonic-frr/Makefile had steps to apply stg (and other) patches. Now a checkout to the original branch is done at the end, and other cleanup is done at the start.

As side-effect, frr now also gets "-sonic" in the version, which is a nice way to see that it's different from the upstream frr.

#### How to verify it

Run:

    make configure PLATFORM=broadcom

    make BLDENV=bookworm -f Makefile.work \
      target/debs/bookworm/frr_8.5.4-sonic-0_amd64.deb

Observe how ./src/sonic-frr/frr is unchanged afterwards. The patched version is still available in a stored stg_temp branch.

#### Which release branch to backport (provide reason below if selected)


^- also relevant for 202405 branch (needs slight change)

#### Tested branch (Please provide the tested image version)

master
